Add warning when accessing non-sandboxed console in production

leeky opened the pull request:
**This is based on [my recent work](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/1074/files#diff-2b40f90adf6d158575c688f23824dd14f1435db009a32e86e5ebea2e0e597c34) improving the developer experience on RODA.**

If you have multiple Rails consoles open, it's often difficult to know whether you're connected to a local console or to production, which risks mistakes being made.

This change makes Pry the default Rails console in all environments, and additionally, when starting a production console without the `--sandbox` switch, show an obvious warning banner to the user:

![Screenshot 2021-05-04 at 13 37 04](https://user-images.githubusercontent.com/3166/117004317-d5398a80-acdd-11eb-93b4-306c5e0aaae1.png)

Additionally, the prompt shows "writable" or "sandbox" on every line, e.g.

![Screenshot 2021-05-04 at 13 40 35](https://user-images.githubusercontent.com/3166/117004755-56911d00-acde-11eb-9cfa-4219fe5eb9b2.png)

You can customise the environment name shown in the warning message and prompts by overriding the `PRODUCTION_ENV` environment variable (e.g. 'staging'):

![Screenshot 2021-05-04 at 13 43 06](https://user-images.githubusercontent.com/3166/117005016-ac65c500-acde-11eb-9cdb-537c91bcc3a2.png)

tahb reviewed (approved):
Very nice indeed. Looks reasonable to me and works when I twiddle it so it runs on my machine 👍

pezholio reviewed (approved):
This is ace, and can confirm it's been very useful for us on RODA

lozette reviewed (approved):
Clever stuff!!

tahb committed:
Add warning when accessing console in production

If you have multiple Rails consoles open, it's often difficult to know
whether you're connected to a local console or to production, which
risks mistakes being made.

This change makes Pry the default Rails console in all environments, and
additionally, when starting a production console without the `--sandbox`
switch, show an obvious warning banner to the user.

Additionally, the prompt shows "writable" or "sandbox" on every line.

You can customise the environment name shown in the warning message and
prompts by overriding the `PRODUCTION_ENV` environment variable (e.g.
'staging').

Co-authored-by: Robert Lee-Cann <leeky@leeky.org.uk>
